internal/commands/user/flags.go

```diff
@@ -35,5 +51,101 @@ var (
 		providerTypeGoogle,
 		providerTypeAnonymous,
 		providerTypeCustom,
+		providerTypeApple,
+		providerTypeCustomFunction,
+	}
+)
+
+// statusType enumset
+type statusType string
+
+// String returns the status type display
+func (st statusType) String() string { return string(st) }
+
+// Type returns the statusType type
+func (st statusType) Type() string { return "string" }
+
+// Set validates and sets the status type value
+func (st *statusType) Set(val string) error {
+	newStatusType := statusType(val)
+
+	if !isValidStatusType(newStatusType) {
+		return errInvalidStatusType
+	}
+
+	*st = newStatusType
+	return nil
+}
+
+// set of statusType(s)
+const (
+	statusTypeEmpty     statusType = ""
+	statusTypeConfirmed statusType = "confirmed"
+	statusTypePending   statusType = "pending"
+)
+
+// statusType error msg
```

Comment:
[nit] `// set of known user errors` (and consolidate the other errors into this `var` block)